src/shutdown/v1/PendleShutdownStrategy.sol

```diff
@@ -0,0 +1,196 @@
+        `>>%%%%>>, `:::::`%%%%%%%%%>>. ;;>%%%%%%                    ;>%,
+         `>>%%%%>>, :::::::`>>>%%%%>>> ;;>%%%%%'                     ;>%,
+          `>>%%%%>>,::::::,>>>>>>>>>>' ;;>%%%%%                       ;%%,
+            >>%%%%>>,:::,%%>>>>>>>>'   ;>%%%%%.                        ;%%
+             >>%%%%>>``%%%%%>>>>>'     `>%%%%%%.
+             >>%%%%>> `@@a%%%%%%'     .%%%%%%%%%.
+             `a@@a%@'    `%a@@'       `a@@a%a@@a
+ */
+
+import "src/shutdown/v1/BaseShutdownStrategy.sol";
+
+import {IVault} from "src/interfaces/IVault.sol";
+import {IStrategy} from "src/interfaces/IStrategy.sol";
+import {IPendleMarket} from "src/interfaces/IPendleMarket.sol";
+import {ILiquidityGauge} from "src/interfaces/ILiquidityGauge.sol";
+import {ReentrancyGuard} from "solady/src/utils/ReentrancyGuard.sol";
+
+/// @notice Strategy contract, supporting Shutdown for Pendle markets.
+contract PendleShutdownStrategy is Ownable2Step, BaseShutdownStrategy, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    /// @notice Address of the strategy.
+    /// @dev It contains most of the storage of the strategy.
+    address public constant STRATEGY = 0xA7641acBc1E85A7eD70ea7bCFFB91afb12AD0c54;
+
+    /// @notice Address of the PENDLE token
+    address public constant PENDLE = 0x808507121B80c02388fAd14726482e061B8da827;
+
+    /// @notice Error when the PENDLE mint failed
+    error MINT_FAILED();
+
+    /// @notice Error when the PENDLE transfer failed
+    error TRANSFER_FAILED();
+
+    /// @notice Error when the caller is not the vault
+    error ONLY_VAULT();
+
+    /// @notice Event when PENDLE is claimed
+    event Claimed(address indexed token, uint256 amount);
+
+    /// @notice Event when a token is withdrawn
+    event Withdrawn(address indexed token, uint256 amount);
+
+    /// @notice Modifier to ensure only the vault can call certain functions
+    modifier onlyVault() {
+        if (!IStrategy(STRATEGY).vaults(msg.sender)) revert ONLY_VAULT();
+        _;
+    }
+
+    /// @notice Constructor to initialize the shutdown strategy
+    /// @param _locker Address of the locker contract
+    /// @param _gateway Address of the gateway contract
+    /// @param _governance Address of the governance contract
+    constructor(address _locker, address _gateway, address _governance)
+        BaseShutdownStrategy(_locker, _gateway, _governance)
+    {}
+
+    /// @dev Reproduces the claim function of the STRATEGY contract and shuts down the gauge.
+    /// @param _token Address of the Pendle market token
+    function claim(address _token) external nonReentrant {
+        if (isShutdown[_token]) revert SHUTDOWN();
+
+        /// 1. Get reward tokens and snapshot balances
+        address[] memory rewardTokens = IPendleMarket(_token).getRewardTokens();
+        uint256[] memory balancesBefore = new uint256[](rewardTokens.length);
+        for (uint8 i; i < rewardTokens.length;) {
```

Comment:
⛽ Minor gas optimization, but you can save a few gas by storing the length of `rewardTokens` in the stack using a type variable. It will save up to length-1 MLOAD.

```suggestion
	uint256 nbrRewardToken = rewardTokens.length;
	for (uint8 i; i < nbrRewardToken;) {
```

Reply:
In fact, storing the length in the stack is even more interesting as you need the length again line 120.